isis/src/control/apps/findfeatures/findfeatures.cpp

```diff
@@ -88,12 +90,69 @@ namespace Isis{
       return;
   }
 
+  /**
+   * @brief Loads train images and applies geometry if provided
+   *
+   * This function loads a trainer image and optionally applies geometry using
+   * a FastGeom transform.
+   *
+   * A MatchImage is added to the MatchMaker as a trainer image. YOU SHOULD NOT
+   * USE THIS FUNCTION TO SET THE QUERY IMAGE. Once loaded, a FastGeom is
+   * optionally applied to the image before adding it to the matcher.
+   *
+   * Errors are trapped here if the file cannot be loaded or the geometry proess
+   * in FastGeom fails. In this case, false is returned without adding the image
+   * to the matcher.
+   *
+   * @author 2021-10-29 Kris J. Becker
+   *
+   * @param matcher    The source reponsible for managing the matcher process
+   * @param trainfile  File name to load
+   * @param fastgeom   Apply FastGeom if provded. Errors will return false with
+   *                      no adding to the matcher
+   *
+   * @return bool
+   */
+  static bool load_train_with_geom(MatchMaker &matcher, const QString trainfile,
+                                  QDebugStream &logger,
+                                  const FastGeom *fastgeom = 0) {
+
+    bool isGood = false;  // If anything goes wrong...
+    try {
+      MatchImage t_image = MatchImage( ImageSource(trainfile) );
+
+      // Compute a FastGeom transform if requested
+      if ( fastgeom ) {
+        // This will typically throw an exception for bad geometry.
+        //  Will return false with no addition to matcher.
+
+        // Note if FastGeom is successful, it adds a transform on
+        // the trainer that computes geometric relationships with
+        // the query image.
+        fastgeom->apply(matcher.query(), t_image, QLogger(logger));
+      }
+
+      // If alls good...expection should be thrown above otherwise
+      matcher.addTrainImage( t_image );
+      isGood = true;
+    }
+    catch ( IException &ie ) {
+      isGood = false;
```

Comment:
Why not just return false here and return true at the end?

Reply:
Ok